added initial karma config & test for browser testing

AndreWillomitzer opened the pull request:
Hi there @kibertoad I set up the initial Karma config file and a basic test :) 

However, when I try to run the test command it doesn't let me because of Docker (which I kind of suspected). 

![image](https://user-images.githubusercontent.com/36431357/144158244-ee20bf37-c139-4fc0-bcd4-0f41ce269d6d.png)

AndreWillomitzer commented:
Alright. Got the Docker and tests set up. Does anyone happen to know what could be happening with the pre-processor? 

![image](https://user-images.githubusercontent.com/36431357/144754575-8dfb0d23-2ad8-48ff-85b7-71dffe0e3a35.png)

@kibertoad anticipated the Undici error because that was why he wanted me to make the browser tests in the first place. I'm more wondering about the pre-processor error because I included 'lib' folder in my karma config.

AndreWillomitzer commented:
Hi there hope you are well @lbdremy. I have been working with Igor Savin on his tool toad-scheduler. He mentioned the tests may not be browser compatible so I wanted to inspect why. I have gotten Docker up and running and I seem to be stuck at this "esModuleInterop" error. I tried looking around and set the flag in tsconfig but it still gives the same error about only default imports allowed.

Do you perhaps have any ideas about what could be the issue?

Thanks and nice to meet you :D

kibertoad commented:
@AndreWillomitzer Try using named import instead of a default one.
Meaning `const {something} = require('something')` instead of `const something = require('something')`

AndreWillomitzer commented:
@kibertoad one step closer boss man: 
![image](https://user-images.githubusercontent.com/36431357/144760124-a55f2008-8892-4316-bf48-0319a1968a9c.png)

AndreWillomitzer commented:
@kibertoad I could get rid of the ProxyAgent one but it requires me going into index.d.ts in the node_modules folder and changing the import statement like I did for the other one. However, this change is untracked. How can I add this to the commits?

After I do so, it's just the headers error left.

AndreWillomitzer commented:
@kibertoad I managed to change it by editing gitignore and then committing and then it worked but I am not sure if this will cause issues in the future.

kibertoad commented:
@AndreWillomitzer You should never commit node_modules, under any circumstances. Those are resolved dynamically from defined dependency list.
Which part of undici types did you need to change to make this work?

AndreWillomitzer commented:
> @AndreWillomitzer You should never commit node_modules, under any circumstances. Those are resolved dynamically from defined dependency list. Which part of undici types did you need to change to make this work?

@kibertoad I changed the `types.ts` import declaration. I changed it to this:
`import Dispatcher = require('undici/types/dispatcher');`
And the only way to get rid of the other Undici error was to do into node_modules and change the index.d.ts to use a new import statement just like that one using require(). The old import was giving errors just like Dispatcher was.

Not sure what to do about that which is why I changed the node_modules thing (sorry if that upset ya).

AndreWillomitzer commented:
@kibertoad Do you happen to know another work around for getting rid of the "esModuleInterop" error? Because the first one I changed was fine in types.ts but that's also not a node_modules file. So when it comes to the other one I don't know how to fix it other than that same solution which you did say not to do because it will commit node_modules. 

Once I changed those import statements it even got rid of the error about file pre-processing.

The issue is that now when I run the solr tests without Karma, it tells me something weird about "figc" not being a function and even changing that import statement doesn't work.
![image](https://user-images.githubusercontent.com/36431357/144878126-3d643ddd-62ac-4079-a60b-8bb7ec0b3ed9.png)

kibertoad commented on README.md:
```diff
@@ -85,6 +85,19 @@ If you want to execute them on oldest Solr version supported, run:
+#### Windows 10 Home
+
+If you have a version of Windows 10 Home, a little more effort is needed. By default, Hyper-V is not able to be turned on for Home.
+**The steps I found to work:**
```
```suggestion
**The steps that seem to work:**
```

kibertoad commented on README.md:
```diff
@@ -85,6 +85,19 @@ If you want to execute them on oldest Solr version supported, run:
+
+#### Windows 10 Home
+
+If you have a version of Windows 10 Home, a little more effort is needed. By default, Hyper-V is not able to be turned on for Home.
```
```suggestion
If you have a version of Windows 10 Home, a little more effort is needed. By default, Hyper-V cannot be turned on for Home edition.
```
